qa/os/src/test/java/org/elasticsearch/packaging/test/PluginCliTests.java

```diff
@@ -0,0 +1,128 @@
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assume.assumeTrue;
+
+public class PluginCliTests extends PackagingTestCase {
+
+    private static final String EXAMPLE_PLUGIN_NAME = "custom-settings";
+    private static final Path EXAMPLE_PLUGIN_ZIP;
+    static {
+        // re-read before each test so the plugin path can be manipulated within tests
+        EXAMPLE_PLUGIN_ZIP = Paths.get(System.getProperty("tests.example-plugin"));
+    }
+
+    @Before
+    public void filterDistros() {
+        assumeTrue("no docker", distribution.packaging != Distribution.Packaging.DOCKER);
+    }
+
+    @FunctionalInterface
+    public interface PluginAction {
+        void run(Shell.Result installResult) throws Exception;
+    }
+
+    private Shell.Result assertWithPlugin(Installation.Executable pluginTool, Path pluginZip, String pluginName, PluginAction action)
+        throws Exception {
+        Shell.Result installResult = pluginTool.run("install --batch \"" + pluginZip.toUri().toString() + "\"");
+        action.run(installResult);
+        return pluginTool.run("remove " + pluginName);
+    }
+
+    private void assertWithExamplePlugin(PluginAction action) throws Exception {
+        assertWithPlugin(installation.executables().pluginTool, EXAMPLE_PLUGIN_ZIP, EXAMPLE_PLUGIN_NAME, action);
+    }
+
+    public void test10Install() throws Exception {
+        install();
+    }
+
+    public void test20SymlinkPluginsDir() throws Exception {
+        Path pluginsDir = installation.plugins;
+        Path stashedPluginsDir = createTempDir("stashed-plugins");
+
+        Files.delete(stashedPluginsDir); // delete so we can replace it
+        Files.move(pluginsDir, stashedPluginsDir);
+        Path linkedPlugins = createTempDir("symlinked-plugins");
+        Platforms.onLinux(() -> sh.run("chown elasticsearch:elasticsearch " + linkedPlugins.toString()));
+        Files.createSymbolicLink(pluginsDir, linkedPlugins);
+        assertWithExamplePlugin(installResult -> {
+            assertWhileRunning(() -> {
+                final String pluginsResponse = makeRequest(Request.Get("http://localhost:9200/_cat/plugins?h=component")).strip();
+                assertThat(pluginsResponse, equalTo(EXAMPLE_PLUGIN_NAME));
+
+                String settingsPath = "_cluster/settings?include_defaults&filter_path=defaults.custom.simple";
+                final String settingsResponse = makeRequest(Request.Get("http://localhost:9200/" + settingsPath)).strip();
+                assertThat(settingsResponse, equalTo("{\"defaults\":{\"custom\":{\"simple\":\"foo\"}}}"));
+            });
+        });
+
+        Files.delete(pluginsDir);
+        Files.move(stashedPluginsDir, pluginsDir);
+    }
+
+    public void test21CustomConfDir() throws Exception {
+        withCustomConfig(confPath -> assertWithExamplePlugin(installResult -> {}));
```

Comment:
I don't remember why some of them booted ES and asserted that the example plugin looks good and some didn't. I think what you have here is fine though, just asserting that the installer didn't blow up is probably good enough in these cases so long as we have a single test that asserts that it did indeed install the plugin in all the way. And you have that above.

:+1: